All my changes to the rst files.

Krebrov001 opened the pull request:
The new version of my changes to the rst files, after rebasing on top of `dev` branch.

Krebrov001 committed:
Edit the python docstrings for the `geopmpy.7`

The documentation for the **Programming Interface** section in the
`geopmpy.7` is from the python docstrings in the `scripts/geopmpy/*.py`
files. This commit updates the docstrings to include rst code in them.

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopm_report.7.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopm_sched.3.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopm_time.3.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopmpy.7.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopmlaunch.1.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopmadmin.1.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopmagent.1.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopmbench.1.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopmctl.1.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopmendpoint.1.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

dannosliwcd commented on scripts/geopmpy/io.py:
```diff
@@ -221,13 +219,13 @@ class IndexTracker(object):
+    ``(<GEOPM_VERSION>, <PROFILE_NAME>, <AGENT_NAME>, <NODE_NAME>)``
 
-    If the tuple not contained in the _run_outputs dict, it is
+    If the tuple not contained in the ``_run_outputs`` dict, it is
```
```suggestion
    If the tuple is not contained in the ``_run_outputs`` dict, it is
```

Krebrov001 committed:
Fix the issues with rebase.

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

dannosliwcd commented on scripts/geopmpy/launcher.py:
```diff
@@ -5,11 +5,11 @@
-is through the geopmlauncher(1) command line executable which calls
-the geopmpy.launcher.main() function.  See the geopmlauncher(1) man
+is through the :doc:`geopmlaunch(1) <geopmlaunch.1>` command line executable which calls
+the ``geopmpy.launcher.main()`` function.  See the :doc:`geopmlaunch(1) <geopmlaunch.1>` man
```
Instead of using a preformatted span for main(), this could alternatively be made into a link to the documentation for main(). You can xref to python functions with
```
:py:func:`some.function`
```

See https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html?highlight=%3Apy%3Ameth#role-py-func

dannosliwcd commented on scripts/geopmpy/io.py:
```diff
@@ -795,7 +791,7 @@ def get_field(self, raw_data, key, units=''):
 class RawReportCollection(object):
     '''
-    Used to group together a collection of related RawReports.
+    Used to group together a collection of related ``RawReport``\ s.
```
Similar to another comment about `:py:func:` you could use `:py:class:` here so that readers of the documentation can just click on `RawReport` to get to its documentation.

dannosliwcd commented on scripts/geopmpy/launcher.py:
```diff
@@ -1138,7 +1138,7 @@ def affinity_option(self, is_geopmctl):
     """
     Launcher derived object for use with Open MPI project launch
-    application mpiexec.
+    application ``mpiexiec``.
```
```suggestion
    application ``mpiexec``.
```

dannosliwcd commented on scripts/geopmpy/launcher.py:
```diff
@@ -1298,7 +1298,7 @@ def affinity_option(self, is_geopmctl):
     def node_list_option(self):
         """
-        Returns a list containing the --host option for mpiexiec.
+        Returns a list containing the ``--host`` option for ``mpiexiec``.
```
```suggestion
        Returns a list containing the ``--host`` option for ``mpiexec``.
```

dannosliwcd commented on scripts/geopmpy/launcher.py:
```diff
@@ -1640,19 +1640,19 @@ def preload_option(self):
+    Main routine used by ``geopmlaunch`` wrapper executable.
     This function creates a launcher from the factory and
-    calls the run method.  If help was requested on the command line
+    calls the ``run()`` method.  If help was requested on the command line
```
run() could also be a python reference link.

Krebrov001 replied:
Thanks for teaching me how to do the python reference links.

dannosliwcd commented on service/docs/source/geopmagent.1.rst:
```diff
@@ -40,16 +44,16 @@ Options
 List all available agents on the system:
 
-.. code-block::
+.. code-block:: bash
```
Maybe use `console` instead of `bash` since this block has a mix of input (prefixed with `$`) and command output.

dannosliwcd commented on service/docs/source/geopmagent.1.rst:
```diff
@@ -58,28 +62,28 @@ List all available agents on the system:
 budget:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmagent.1.rst:
```diff
@@ -58,28 +62,28 @@ List all available agents on the system:
 budget and other policies set to default:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmagent.1.rst:
```diff
@@ -89,7 +93,7 @@ Create policy for monitor agent which does not require any policies.
 specifying ``--geopm-agent`` and ``--geopm-policy`` are optional.
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmagent.1.rst:
```diff
@@ -58,28 +62,28 @@ List all available agents on the system:
+Get ``power_balancer`` agent policy and sample names:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmendpoint.1.rst:
```diff
@@ -51,15 +57,15 @@ Examples
 List all endpoints on a system when none are open:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmendpoint.1.rst:
```diff
@@ -51,15 +57,15 @@ Examples
 attach:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmendpoint.1.rst:
```diff
@@ -70,36 +76,36 @@ attach:
 yet attached:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmendpoint.1.rst:
```diff
@@ -70,36 +76,36 @@ attach:
+``power_balancer`` agent has attached:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmendpoint.1.rst:
```diff
@@ -70,36 +76,36 @@ attach:
 node power budget:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmendpoint.1.rst:
```diff
@@ -70,36 +76,36 @@ attach:
+Sample from balancing agent with endpoint ``"job-321"``:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmendpoint.1.rst:
```diff
@@ -108,9 +114,9 @@ Sample from balancing agent with endpoint "job-321":
+Destroy endpoints ``"job-123"`` and ``"job-321"``:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block:: console
```

dannosliwcd commented on service/docs/source/geopmlaunch.1.rst:
```diff
@@ -374,13 +377,15 @@ GEOPM Options
                     clean up after an aborted job is:
 
-                    ``$ test -n "$GEOPM_SHMKEY" && rm -f /dev/shm${GEOPM_SHMKEY}* || rm -f /dev/shm/geopm-shm*``
+                    .. code-block:: bash
```
```suggestion
                    .. code-block:: console
```

(or drop the leading `$` and keep `bash`)

dannosliwcd commented on service/docs/source/geopmlaunch.1.rst:
```diff
@@ -584,7 +589,7 @@ the ``geopmctl`` binary.
 GEOPM configuration files exist:
 
-.. code-block::
+.. code-block:: bash
```
```suggestion
.. code-block::
```
This is just a list of file paths, so no need for bash highlighting

dannosliwcd commented on service/docs/source/geopmpy.7.rst:
```diff
@@ -20,9 +20,9 @@ In addition, there is 1 infrastructure module provided:
 ^^^^^
 This module provides tools for parsing and encapsulating report and trace data
-into either simple structures or Pandas DataFrames.  It can be used to parse
+into either simple structures or **Pandas** ``DataFrame``\ s.  It can be used to parse
```
Could alternatively link to `pandas.DataFrame` using a `:py:class:` reference. Pandas is part of our intersphinx list so it should automatically figure out the right URL.

Krebrov001 replied:
> Could alternatively link to `pandas.DataFrame` using a `:py:class:` reference. Pandas is part of our intersphinx list so it should automatically figure out the right URL.

This is in a `*.rst` file though, not in a python docstring.

dannosliwcd replied:
Here's the Sphinx documentation about how to use that feature in rst files: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html?highlight=%3Apy%3Ameth#role-py-func

Krebrov001 committed:
Add `add_function_parentheses = True` to the `conf.py`

to get Sphinx to generate the reference links with parentheses at the
ends of functions.

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Addressing review updates

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>
